### PR TITLE
Add snippet includes feature. Introduce new snippet UX/UI.

### PR DESCRIPTION
**Pull request by pikinier20**

### CODE REVIEW

File: scaladoc-js/resources/scaladoc-searchbar.css [hunk 1]
```diff
@@ -124,34 +124,201 @@
...
+    padding-top: 10px;
+    font-size: 0.75em;
+}
+
+.snippet-meta .snippet-label {
+    font-weight: bold;
+}
+
+.snippet .buttons {
+    --icon-size: 24px;
```
Comment (KacperFKorban) [inline scaladoc-js/resources/scaladoc-searchbar.css:142]: Wouldn't it be better to make it `16px` as a default? Those big icons don't really fit the rest of the site.

Comment (KacperFKorban) [reply scaladoc-js/resources/scaladoc-searchbar.css:142]: If so, the size of the slider should be a bit smaller as well.

File: scaladoc-testcases/src/tests/snippetTestcase2.scala [hunk 2]
```diff
@@ -19,7 +19,7 @@ trait Quotes2[A] {
      val z: zModule = ???
      trait zModule {
       /**
-        * SNIPPET(OUTERLINEOFFSET:25,OUTERCOLUMNOFFSET:10,INNERLINEOFFSET:7,INNERCOLUMNOFFSET:6)
+        * SNIPPET(OUTERLINEOFFSET:25,OUTERCOLUMNOFFSET:10,INNERLINEOFFSET:9,INNERCOLUMNOFFSET:6)
```
Comment (KacperFKorban) [inline scaladoc-testcases/src/tests/snippetTestcase2.scala:22]: Those shouldn't be rendered, right?

**Review by KacperFKorban — CHANGES_REQUESTED** [review]
Looks great overall. I just found some minor bugs and suggestions.
- [x] There seems to be a clash between extending members and snippet actions. 
Peek 2021-08-02 11-23
- [ ] Some garbage is left when extending included snippets (`case1`)
image

File: scaladoc-testcases/src/tests/snippetTestcase2.scala (hunk 2 above)
Comment (pikinier20) [reply scaladoc-testcases/src/tests/snippetTestcase2.scala:22]: This is information for tests to assert good positions of errors but it's also a part of docstring so I think it'll be rendered.

**Review by KacperFKorban — APPROVED** [review]
LGTM. We should probably create an issue for the formatting problem.

**Pull request merged**